packages/fuels/src/cli/utils/checkForAndDisplayUpdates.ts

```diff
@@ -0,0 +1,43 @@
+import { versions, gt, eq } from '@fuel-ts/versions';
+
+import { warn, log } from './logger';
+
+export const getLatestFuelsVersion = async () => {
+  const response = await fetch('https://registry.npmjs.org/fuels/latest');
+  const data = await response.json();
+  return data.version as string;
+};
+
+export const checkForAndDisplayUpdates = async () => {
+  try {
+    const { FUELS: userFuelsVersion } = versions;
+
+    const latestFuelsVersion = await Promise.race<string | undefined>([
+      new Promise((resolve) => {
+        setTimeout(resolve, 3000);
+      }),
+      getLatestFuelsVersion(),
+    ]);
```

Comment:
@Dhaiwat10 I couldn't review this one final time before the merge, but I will leave notes here.

As I mentioned in my previous [comment](https://github.com/FuelLabs/fuels-ts/pull/3108#issuecomment-2346355732), I really think we should use a [local cache](https://github.com/node-cache/node-cache) here.

Requesting npm versions for every CLI command we run seems unnecessary.

A TTL of 6 hours should do wonders already. Thoughts?

Reply:
@arboleya my bad, I missed that point of yours. I have filed an issue and will do a follow-up PR soon: https://github.com/FuelLabs/fuels-ts/issues/3177

Reply:
Thanks.